### ISSUE TITLE
How to create node with entity field

### ISSUE THREAD

**Issue by fufu7755** [signalpoint/angular-drupal#39]
@signalpoint Thank you for helping me solve the last problem. I have found the answer.

I have a new problem, when I create node with entity field always gave error, I think it should be the format problem.

`var node = {
  type: 'article',
  title: 'Hello world',
  language: 'und',
  body: { und: [ { value: 'How are you?' }] }
};`

I change them to 

`var node = {
  type: 'article',
  title: 'Hello world',
  language: 'und',
  fiele_engtity: { und: [ { entity_id : *** }] }
};`

Pleas give me some tips. thank you very much.
Hopefully it will not take you too much time.

**Comment by signalpoint** [comment]
> I have a new problem, when I create node with entity field always gave error, I think it should be the format problem.

@fufu7755 What is the error?

**Comment by kentr** [comment]
@fufu7755 Is this from your actual code?

``` javascript
var node = { type: 'article', title: 'Hello world', language: 'und', fiele_engtity: { und: [ { entity_id : *** }] } };
```

What happens if you change `fiele_engtity` to `field_entity`?

**Comment by signalpoint** [comment]
@fufu7755 I agree with @kentr and it's likely you'll need to use `target_id` instead of `entity_id` like so:

```
var node = {
  type: 'article',
  title: 'Hello world',
  language: 'und',
  field_entity: { und: [ { target_id : '... (123)' }] }
};
```

**Comment by fufu7755** [comment]
I use 
`var node = {
  type: 'article',
  title: 'Hello world',
  language: 'und',
  field_entity: { und: [ { target_id : 123 }] }
};`

`var node = {
  type: 'article',
  title: 'Hello world',
  language: 'und',
  field_entity: { und: [ { target_id : '123' }] }
};`

`var node = {
  type: 'article',
  title: 'Hello world',
  language: 'und',
  field_entity: { und: [ { target_id : 'node(123)' }] }
};`

still return error message.

**Comment by signalpoint** [comment]
@fufu7755 What's the error message?

**Comment by fufu7755** [comment]
Message is "Not Acceptable : An illegal choice has been detected. Please contact the site administrator."

**Comment by fufu7755** [comment]
![wechatimg2](https://cloud.githubusercontent.com/assets/7938785/19954416/21c6153e-a1b3-11e6-98c3-303a770d48af.jpeg)

**Comment by signalpoint** [comment]
@fufu7755 It should also mention what field is causing the problem. What is that field? Then what is the **Widget** set to for that particular field on your **Manage fields** page? The way in which you send the the data is sometimes dependent on the widget selected for the field.

As an example, if you're using an autocomplete widget for your entity reference field, then you need to send a value like this:

```
field_entity: { und: [ { target_id : '... (123)' }] }
```

The `...` and the space are important, the placement of the parenthesis are important, and the integer value must lie within the parenthesis.

**UPDATE**

As I can see from your screenshot, the field I am asking questions about it the `field_record_pid` field. What type of field is this? What module powers it? What widget is it using? Is it one, or more, or unlimited values, etc?

**Comment by fufu7755** [comment]
Thank you very much. I see what you mean. Using the autocomplete widget, the code format is
`field_entity: { und: [ { target_id : '... (123)' }] }`
If I use the Check box or select list, then what is the format of the code？

My English is not good. Thank you so much for your patience.

**Comment by fufu7755** [comment]
If I set entity field's **Widget** with **Check box**, then what is code format?
Thanks

**Comment by signalpoint** [comment]
@fufu7755 I don't know what the format is, I'm guessing it is just this:

```
field_entity: { und: [ { target_id : '123' }] }
```

In cases like this, I usually open up Firefox Poster and try many different combinations of JSON, until it properly saves.

However, my question still stands:

> As I can see from your screenshot, the field I am asking questions about it the field_record_pid field. What type of field is this? What module powers it? What widget is it using? Is it one, or more, or unlimited values, etc?

**Comment by fufu7755** [comment]
This field is node entity field and use autocomplete as Widget, just one value.

![wechatimg1](https://cloud.githubusercontent.com/assets/7938785/19967433/247e1be8-a20b-11e6-9945-21a0b3a54a0e.jpeg)

**Comment by fufu7755** [comment]
@signalpoint Question again, If I want to work with more or unlimited values, how to write code.
Thanks again.

**Comment by signalpoint** [comment]
@fufu7755 If it's an autocomplete, then definitely use the format previosly discussed. For multiple values you need [this patch](https://www.drupal.org/node/2224803), and this format:

```
field_entity: { und: [
  { target_id : '... (123)' }],
  { target_id : '... (456)' }],
}
```

**Comment by fufu7755** [comment]
Thank you very much.
